pylint/checkers/variables.py

```diff
@@ -976,267 +989,261 @@ def visit_assignname(self, node: nodes.AssignName) -> None:
-                    # Also do not consume class name
-                    # (since consuming blocks subsequent checks)
-                    # -- quit
-                    break
+        (
+            maybe_before_assign,
+            annotation_return,
+            use_outer_definition,
+        ) = self._is_variable_violation(
+            node,
+            defnode,
+            stmt,
+            defstmt,
+            frame,
+            defframe,
+            base_scope_type,
+            is_recursive_klass,
+        )
 
-                (
-                    maybee0601,
-                    annotation_return,
-                    use_outer_definition,
-                ) = self._is_variable_violation(
-                    node,
-                    defnode,
-                    stmt,
-                    defstmt,
-                    frame,
-                    defframe,
-                    base_scope_type,
-                    recursive_klass,
-                )
+        if use_outer_definition:
+            return (VariableVisitConsumerAction.CONTINUE, None)
 
-                if use_outer_definition:
-                    continue
+        if (
+            maybe_before_assign
+            and not utils.is_defined_before(node)
+            and not astroid.are_exclusive(stmt, defstmt, ("NameError",))
+        ):
 
-                if (
-                    maybee0601
-                    and not utils.is_defined_before(node)
-                    and not astroid.are_exclusive(stmt, defstmt, ("NameError",))
-                ):
+            # Used and defined in the same place, e.g `x += 1` and `del x`
+            defined_by_stmt = defstmt is stmt and isinstance(
+                node, (nodes.DelName, nodes.AssignName)
+            )
+            if (
+                is_recursive_klass
+                or defined_by_stmt
+                or annotation_return
+                or isinstance(defstmt, nodes.Delete)
+            ):
+                if not utils.node_ignores_exception(node, NameError):
 
-                    # Used and defined in the same place, e.g `x += 1` and `del x`
-                    defined_by_stmt = defstmt is stmt and isinstance(
-                        node, (nodes.DelName, nodes.AssignName)
-                    )
-                    if (
-                        recursive_klass
-                        or defined_by_stmt
-                        or annotation_return
-                        or isinstance(defstmt, nodes.Delete)
+                    # Handle postponed evaluation of annotations
+                    if not (
+                        self._postponed_evaluation_enabled
+                        and isinstance(
+                            stmt,
+                            (
+                                nodes.AnnAssign,
+                                nodes.FunctionDef,
+                                nodes.Arguments,
+                            ),
```

Comment:
We could create a variable for this for better formatting / understanding thanks to the name of the new variable.

Reply:
Any suggestions for a name?

Reply:
I'm wondering too. variable definition nodes ?

Reply:
Not sure if it increase readability btw. Because then the type variables would be at the top of the file and you would need to scroll to see it.
Perhaps we should leave it as is? Not sure here...

Reply:
Fair enough :)